src/ray/common/ray_object.h

```diff
@@ -109,9 +100,35 @@ class RayObject {
   int64_t CreationTimeNanos() const { return creation_time_nanos_; }
 
  private:
+  void Init(const std::shared_ptr<Buffer> &data, const std::shared_ptr<Buffer> &metadata,
```

Comment:
I am using this because the constructor doesn't seem to allow us to have other constructor within it. For example,

```
Construct(A) {
    Construct(A, B) 
}
```
This didn't seem to possible

Reply:
hmm i think you can. you can use 
```
Ctor(A): Ctor(A, B) {
  ...
}
```
is this what you are looking for?

Reply:
This is not working because I wanted to add some logic before calling the constructor Ctor(A, B).

Reply:
something like this (not completely the same)
```cpp
Ctor(A) {
    if (logic()) {
        Ctor(A, B)
    } else {
        Ctor(A, C)
    }
}
```

If we cannot write this logic, I need to change lots of part of code right now..